lib/tasks_private/lib/aws_instance_data_parser.rb

```diff
@@ -0,0 +1,336 @@
+# frozen_string_literal: true
+
+class AwsInstanceDataParser
```

Comment:
This class feel overly complex with all of the `concerning`, `memoizable`, a special helper method for memoization, hardcoded constants, delayed parsing, etc.  Can you explain why this is all needed.  In the back of my head, it feels like an awful lot of code when the goal is to take one-or-more jsons and transform to a different json.  That feels like it would be at most it would be a 100 line method, so I feel like I'm missing something.

Reply:
For readability and maintainability.

Reply:
By hardcoded I constants I meant things like https://github.com/ManageIQ/manageiq-providers-amazon/pull/449/files#diff-c1dcfc36a8dc66b1c22b3401ea962b35R43, where there are specific types hardcoded as opposed to discovered from the raw data.  However, it might be a necessary evil if the raw data sources don't have that association.

Reply:
Yes, some data is missing there. 
BTW, maybe we don't need some fields at all?
I have no idea where they used, maybe you can point me on?

Reply:
It's quite possible we don't need a lot of the fields...you'd have to search the parser for what is used out of the instance types data.  When I first built the file, we wanted everything, and since I had to manually scrape anyway, it was in the same place, so I just added it in, because we didn't know what would be used.  Ultimately we used *some* of them, but there are things we definitely don't use.  I would be ok with dropping the unused things.

Reply:
In that case I propose to do that in a separate PR, as it can take much time, and those 2 BZs will still stay unclosed.